File: morphac/constants/test/map_constants_test.cc
#include "gtest/gtest.h"

#include "constants/include/map_constants.h"

namespace {

using morphac::constants::MapConstants;

class MapConstantsTest : public ::testing::Test {
 protected:
  MapConstantsTest() {}

  void SetUp() override {}
};

TEST_F(MapConstantsTest, Members) {
  ASSERT_EQ(MapConstants::EMPTY, 0.);
  ASSERT_EQ(MapConstants::OBSTACLE, -1.);
}

}  // namespace

int main(int argc, char** argv) {
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}

File: morphac/constants/python/test/map_constants_test.py
import pytest

from morphac.constants import MapConstants


def test_attributes():

    assert MapConstants.EMPTY == 0.0
    assert MapConstants.OBSTACLE == -1.0


def test_invalid_setting():

    # The values should not be modifiable.

    with pytest.raises(AttributeError):
        MapConstants.EMPTY = 1.0
    with pytest.raises(AttributeError):
        MapConstants.OBSTACLE = -1.0
